ykrt/pass_finder/try_passes.py

```diff
@@ -66,53 +66,96 @@ def log(logf, s):
 def executable_exists(cmd):
     return shutil.which(cmd) is not None
 
-def get_all_passes(is_prelink):
-    cmd = get_opt_cmd(is_prelink) 
-    p = Popen(cmd, shell=True, stdout=PIPE, close_fds=True)
-    sout, serr = p.communicate() 
-    assert(p.returncode == 0)
+def split_passes(passes_string):
+    parts = []           
+    temp_part = []      
+    paren_stack = []     
+    angle_stack = []    
+
+    i = 0  
+    while i < len(passes_string):
```

Comment:
Since we only `i += 1` in the loop I tihink this is equivalent to:

```python
for c in passes_string:
  ...
```

where `c` is what is currently called `char`.